### PR TITLE
Pbi3 landmark list

### PR DESCRIPTION
**Pull request by skuuzie**
Proposed changes:
- Switch landmark detail page from fragment to activity
- Implement landmark detail repository and it's remote data source dependencies with dagger DI, in accordance to existing DI pattern/style
- Temporarily disable local database cache for landmark detail until further inspection
- Display landmark list on home fragment
- Landmark detail viewmodel is now functional to perform GET request from API

Minor changes:
- Reformat codes with built-in Android Studio reformatter
- Adjustment to UI (positioning, images, fonts, colors, dimens, styles, themes)
- Add a button on post-scan page to test out [Scan -> Landmark detail page] functionality

### CODE REVIEW

File: app/src/main/java/com/dicoding/motour/data/db/EducaTourDB.kt [hunk 1]
```diff
@@ -4,34 +4,33 @@ import androidx.room.Database
...
 import com.dicoding.motour.data.model.landmark.detail.TicketPrice
 import com.dicoding.motour.data.model.landmark.list.Landmark
 
+@TypeConverters(StringListConverter::class)
 @Database(
     entities = [
         ContactInfo::class,
         Landmark::class,
-        LandmarkDetail::class,
+        // LandmarkDetail::class,
```
Comment (IbniShaquille) [inline app/src/main/java/com/dicoding/motour/data/db/EducaTourDB.kt]: remove comment line

File: app/src/main/java/com/dicoding/motour/data/db/EducaTourDB.kt [hunk 2]
```diff
@@ -4,34 +4,33 @@ import androidx.room.Database
...
     version = 1,
     exportSchema = false
 )
-@TypeConverters(StringListConverter::class)
-abstract class EducaTourDB : RoomDatabase(){
+abstract class EducaTourDB : RoomDatabase() {
     abstract fun landmarkDao(): LandmarkDao
-    abstract fun landmarkDetailDao(): LandmarkDetailDao
+
+    // abstract fun landmarkDetailDao(): LandmarkDetailDao
```
Comment (IbniShaquille) [inline app/src/main/java/com/dicoding/motour/data/db/EducaTourDB.kt]: delete  comment line

File: app/src/main/java/com/dicoding/motour/data/repository/landmark/detail/LandmarkDetailRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    override suspend fun getLandmarkById(id: Int): LandmarkDetail? {
+        return getLandmarkFromAPI(id)
+    }
+
+    private suspend fun getLandmarkFromAPI(id: Int): LandmarkDetail? {
+        var landmark: LandmarkDetail? = null
+        try {
+            val response = landmarkDetailRemoteDatasource.getLandmarkDetail(id)
+            Log.d("LandmarkDetailRepositoryImpl", response.raw().toString())
```
Comment (IbniShaquille) [inline app/src/main/java/com/dicoding/motour/data/repository/landmark/detail/LandmarkDetailRepositoryImpl.kt]: for privacy issue, if this function is done, just remove the Logging

File: app/src/main/java/com/dicoding/motour/data/repository/landmark/detail/LandmarkDetailRepositoryImpl.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            val body = response.body()
+            if (body != null) {
+                landmark = body.landmark
+            }
+        } catch (e: Exception) {
+            Log.d("LandmarkDetailRepositoryImpl", e.message.toString())
```
Comment (IbniShaquille) [inline app/src/main/java/com/dicoding/motour/data/repository/landmark/detail/LandmarkDetailRepositoryImpl.kt]: remove this, or you can use Toast

File: app/src/main/java/com/dicoding/motour/presentation/di/core/DataBaseModule.kt [hunk 5]
```diff
@@ -20,10 +20,11 @@ class DataBaseModule {
     @Provides
     fun provideContactInfoDao(educaTourDB: EducaTourDB) = educaTourDB.contactInfoDao()
 
+    /*
     @Singleton
     @Provides
     fun provideLandmarkDetailDao(educaTourDB: EducaTourDB) = educaTourDB.landmarkDetailDao()
-
+    */
```
Comment (IbniShaquille) [inline app/src/main/java/com/dicoding/motour/presentation/di/core/DataBaseModule.kt]: just remove

File: app/src/main/java/com/dicoding/motour/presentation/home/HomeFragment.kt [hunk 6]
```diff
@@ -32,15 +41,61 @@ class HomeFragment : Fragment() {
...
+                        override fun onItemClick(landmark: Landmark) {
+                            Intent(requireContext(), LandmarkDetailActivity::class.java).apply {
+                                putExtra(LandmarkDetailActivity.EXTRA_LANDMARK_ID, landmark.id)
+                            }.run {
+                                startActivity(this)
+                            }
+                        }
+                    })
+
+                    Log.d("MYTAG", landmarkList.toString())
```
Comment (IbniShaquille) [inline app/src/main/java/com/dicoding/motour/presentation/home/HomeFragment.kt]: privacy issue, remove logging if done

File: app/src/main/java/com/dicoding/motour/presentation/home/HomeFragment.kt [hunk 7]
```diff
@@ -32,15 +41,61 @@ class HomeFragment : Fragment() {
...
+            }
+        }
+    }
+
+    private fun setLoading(state: Boolean) {
+        if (state) {
+            binding.progressIndicator.visibility = View.VISIBLE
+        } else {
+            binding.progressIndicator.visibility = View.GONE
+        }
```
Comment (IbniShaquille) [inline app/src/main/java/com/dicoding/motour/presentation/home/HomeFragment.kt]: for the simple mode use this one

binding.progressBar.visibility = if (isLoading) View.VISIBLE else View.GONE

**Comment by IbniShaquille** [comment]
if class StringList is not used, just remove the class

**Comment by IbniShaquille** [comment]
overalll, the code implementation is very good, goodjob!!!

**Comment by skuuzie** [comment]
Refactored the code as per code review. I've also improved the UI for landmark list fragment and added a custom error message for retry mechanism in case of network error

**Comment by IbniShaquille** [comment]
okei, you can merge the PR

**Pull request merged**